edr_server/core/models/extents.py

```diff
@@ -43,6 +43,20 @@ class TemporalExtent(EdrModel["TemporalExtent"]):
     intervals: List[DateTimeInterval] = dataclasses.field(default_factory=list)
     trs: CrsObject = DEFAULT_TRS
 
+    def __post_init__(self):
+        if not isinstance(self.values, List):
+            raise TypeError(f'Expected List of values, received {type(self.values)}')
+        for value in self.values:
+            if not isinstance(value, datetime):
+                raise TypeError(f"Expected all datetime values, received value '{value}' of type {type(value)}")
```

Comment:
Great minds think alike - I actually started with this, but then changed to the separate `for` loop so we could access `value` within the error message.

Reply:
good point!
I think it can be made to work with a walrus assignment?

```python
if not all(isinstance( (invalid_value := value), datetime) for value in self.values):
    raise TypeError(f"Expected all datetime values, received value '{invalid_value}')
```

It works because `all` uses shortcut logic and stops on the first value that doesn't meet the criteria

Reply:
great idea, works brilliantly!